homeassistant/components/media_player/directv.py

```diff
@@ -53,67 +57,244 @@
+        self._receiver_id = receiver_id if receiver_id != host else None
+
+        self.dtv = None
+
+        # Client discovery to be started once HASS is started to ensure
+        # all configured devices have been added first.
+        def client_discovery_startup(event):
+            # Perform a discovery if the main entity was discovered as well or
+            # HASS started.
+            if self._discovered or event:
+                self._discover_directv_client_devices()
+
+            # Schedule discovery to run based on interval.
+            track_time_interval(
+                self._hass, self._discover_directv_client_devices,
+                DEFAULT_CLIENT_DISCOVER_INTERVAL)
+            _LOGGER.debug("%s: Client discovery scheduled for every %s",
+                          self._name, DEFAULT_CLIENT_DISCOVER_INTERVAL)
+
+        # If HASS is already running then start the discovery.
+        # If HASS is not yet running, register for the event before starting
+        # the discovery.
+        if self._hass.is_running:
+            client_discovery_startup(None)
+        else:
+            self._hass.bus.listen_once(
+                EVENT_HOMEASSISTANT_START, client_discovery_startup)
+
+    def _discover_directv_client_devices(self, now=None):
+        """Discover new client devices connected to the main DVR."""
+        known_devices = self._hass.data.get(DATA_DIRECTV)
+        discovered_devices = []
+        dtvs = []
+
+        # Attempt to discover additional RVU units
+        if now:
+            _LOGGER.debug("%s: Scheduled discovery of DirecTV devices on %s",
+                          self._name, self._host)
+        else:
+            _LOGGER.debug("%s: Initial discovery of DirecTV devices on %s",
+                          self._name, self._host)
+
+        _LOGGER.debug("%s: Current known devices: %s",
+                      self._name, known_devices)
+
+        if not self.dtv:
+            from DirectPy import DIRECTV
+            try:
+                self.dtv = DIRECTV(self._host, self._port, DEFAULT_DEVICE)
+            except requests.exceptions.RequestException as ex:
+                # Use uPnP data only
+                _LOGGER.debug("%s: Request exception %s trying to get "
+                              "locations", self._name, ex)
+                self.dtv = None
+
+        if not self.dtv:
+            return
+
+        # If for some reason we did not have a receiver id then retrieve
+        # it now.
+        if not self._receiver_id:
+            receiver_id_resp = None
+            try:
+                receiver_id_resp = self.dtv.get_version()
+            except requests.exceptions.RequestException as ex:
+                _LOGGER.debug("Request exception %s trying to get "
+                              "receiver id for %s", ex, self._name)
+                receiver_id_resp = None
+
+            if not receiver_id_resp:
+                self._receiver_id = receiver_id_resp.get('receiverId')
```

Comment:
This will crash, because you check if `receiver_id_resp` is falsey, and then try to get data from it?